ffmpeg/lpms_ffmpeg.c

```diff
@@ -75,6 +75,16 @@ struct input_ctx {
   // Decoder flush
   AVPacket *first_pkt;
   int flushed;
+  int flushing;
+  // The diff of `packets sent - frames recv` serves as an estimate of
+  // internally buffered packets by the decoder.  We're done flushing when this
+  // becomes 0.
+  uint16_t pkt_diff;
+  // We maintain a count of sentinel packets sent without receiving any
+  // valid frames back, and stop flushing if it crosses SENTINEL_MAX.
+  // FIXME This is needed due to issue #155 - input/output frame mismatch.
```

Comment:
Do you think that once #155 is fixed, we won't need to track `sentinel_count` anymore?

Reply:
I think so because that was the only case for which using only `pkt_diff` failed for me. We can still keep `sentinel_count` around just to be safe.

Reply:
> We can still keep sentinel_count around just to be safe.

The less state we have to explicitly track, the better. But it's fine for now if it maintains the current behavior.

Speaking of which, how was `TestTranscoder_MismatchedEncodeDecode ` failing? The test is set up such that if #155 is ever fixed, then the test will fail; see https://github.com/livepeer/lpms/blob/d722afc1dfa0e7b7102db8b67588a00be6dfd938/ffmpeg/ffmpeg_test.go#L1151-L1154 .

Reply:
The testcase itself was not failing, sorry if that wasn't clear. What happened was that the decoder never returned/exited from flushing mode and went into an infinite loop of sending sentinel packets (because `pkt_diff == 1` and never reached 0)

So now we will not care about `pkt_diff` and force quit if we've sent 5 sentinel packets without getting any valid frames back.

Reply:
Ah gotcha, that also answers the question I had here https://github.com/livepeer/lpms/pull/189#discussion_r423216120 . Thanks!